safe_transaction_service/history/cache.py

```diff
@@ -0,0 +1,200 @@
+    LIST_TRANSFERS_VIEW_CACHE_KEY = "transfersview"
+
+    def __init__(self, cache_tag: str, address: ChecksumAddress):
+        self.redis = get_redis()
+        self.address = address
+        self.cache_tag = cache_tag
+        self.cache_name = self.get_cache_name()
+
+    def get_cache_name(self) -> str:
+        """
+        Calculate the cache_name from the cache_tag and address
+
+        :param cache_tag:
+        :param address:
+        :return:
+        """
+        return f"{self.cache_tag}:{self.address}"
+
+    @cache
+    def is_enabled(self) -> bool:
+        """
+
+        :return: True if cache is enabled False otherwise
+        """
+        if settings.CACHE_VIEW_DEFAULT_TIMEOUT:
+            return True
+        else:
+            return False
+
+    def get_cache_data(self, cache_path: str) -> Optional[str]:
+        """
+        Return the cache for the provided cache_path
+
+        :param cache_path:
+        :return:
+        """
+        if self.is_enabled():
+            logger.debug(f"Getting from cache {self.cache_name}{cache_path}")
+            return self.redis.hget(self.cache_name, cache_path)
+        else:
+            return None
+
+    def set_cache_data(self, cache_path: str, data: str, timeout: int):
+        """
+        Set a cache for provided data with the provided timeout
+
+        :param cache_path:
+        :param data:
+        :param timeout:
+        :return:
+        """
+        if self.is_enabled():
+            logger.debug(
+                f"Setting cache {self.cache_name}{cache_path} with TTL {timeout} seconds"
+            )
+            self.redis.hset(self.cache_name, cache_path, data)
+            self.redis.expire(self.cache_name, timeout)
+        else:
+            logger.warning("Cache txs view is disabled")
+
+    def remove_cache(self):
+        """
+        Remove cache key stored in redis for the provided parameters
+
+        :param cache_name:
+        :return:
+        """
+        logger.debug(f"Removing all the cache for {self.cache_name}")
+        self.redis.unlink(self.cache_name)
+
+
+def cache_txs_view_for_address(
+    cache_tag: str, timeout: int = settings.CACHE_VIEW_DEFAULT_TIMEOUT
+):
+    """
+    Custom cache decorator that caches the view response.
+    This decorator caches the response of a view function for a specified timeout.
+    It allows you to cache the response based on a unique cache name, which can
+    be used for invalidating.
+
+    :param timeout: Cache timeout in seconds.
+    :param cache_name: A unique identifier for the cache entry.
+    """
+
+    def decorator(view_func):
+        @wraps(view_func)
+        def _wrapped_view(request, *args, **kwargs):
+            # Get query parameters
+            query_params = request.request.GET.dict()
+            cache_path = f"{urlencode(query_params)}"
+            # Calculate cache_name
+            address = request.kwargs["address"]
```

Comment:
Maybe `"address"` should be an optional parameter to the decorator, so we can use different keys in the future

Also this can throw an exception if `address` does not exist. It should be: `request.kwargs.get("address")` so you can emit a warning when you evaluate `if address:` later

Reply:
Either way, making `address` as a url path value configurable is just adding an optional parameter (by default `'address'`) to the decorator, something like `caching_value_name`